model_inference_triage_pipeline/run_model_inference.py

```diff
@@ -1,557 +1,379 @@
-
-parser.add_argument("-eco", "--eco-system", type=str, default="openshift",
-                    choices=["openshift", "knative", "kubevirt"],
-                    help="The eco-system to monitor")
-
-parser.add_argument("-device", "--compute-device", type=str, default="gpu",
-                    choices=["gpu", "cpu", "GPU", "CPU"],
-                    help="[Not implemented should work automatically] The backend device to run the AI models on - GPU or CPU")
-
-parser.add_argument("-sec-model", "--security-filter-model", type=str, default="gru",
-                    choices=["gru", "GRU"],
-                    help="The AI Model to use for security filtering - Model 1")
-
-parser.add_argument("-cve-model", "--probable-cve-model", type=str, default="gru",
-                    choices=["gru", "GRU", "bert", "BERT"],
-                    help="The AI Model to use for probable CVE predictions - Model 2")   
-
-parser.add_argument("-s3-upload", "--s3-upload-cves", type=bool, default=False,
-                    choices=[False, True],
-                    help="Uploads inference CSVs to S3 bucket - turn off since bucket works only with my account for now (write access") 
-
-args = parser.parse_args()
-
-DAYS_SINCE_YDAY = args.days_since_yday
-ECO_SYSTEM = args.eco_system.lower()
-COMPUTE_DEVICE = args.compute_device.lower()
-SEC_MODEL_TYPE = args.security_filter_model.lower()
-CVE_MODEL_TYPE = args.probable_cve_model.lower()
-S3_UPLOAD = args.s3_upload_cves
+import pandas as pd
+import torch
+
+from models import run_bert_tensorflow_model, run_gru_cve_model
+from utils import aws_utils as aws
+from utils import bq_client_helper as bq_helper
+from utils import cloud_constants as cc
+from utils.disk_utils import write_output_csv_disk
 
 daiquiri.setup(level=logging.INFO)
 _logger = daiquiri.getLogger(__name__)
+warnings.simplefilter(action="ignore", category=FutureWarning)
+warnings.simplefilter(action="ignore", category=Warning)
+
+
+def main():
+    """The main program logic."""
+    parser = get_argument_parser()
+    args = parser.parse_args()
+
+    DAYS_SINCE_YDAY = args.days_since_yday
+    ECO_SYSTEM = args.eco_system.lower()
+    COMPUTE_DEVICE = args.compute_device.lower()
+    if COMPUTE_DEVICE != "cpu":
+        torch.set_default_tensor_type("torch.cuda.FloatTensor")
+    SEC_MODEL_TYPE = args.security_filter_model.lower()
+    if SEC_MODEL_TYPE != "" and SEC_MODEL_TYPE != "gru":
+        _logger.error("GRU is the only supported security model type.")
+        sys.exit(0)
+    CVE_MODEL_TYPE = args.probable_cve_model.lower()
+    S3_UPLOAD = args.s3_upload_cves
+
+    day_count, start_time, end_time, date_range = setup_dates_for_triage(
+        days_since_yday=DAYS_SINCE_YDAY
+    )
+    df = get_bq_data_for_inference(ECO_SYSTEM, day_count, date_range)
+    run_inference(df, CVE_MODEL_TYPE)
+    triage_results_dir = write_output_csv_disk(
```

Comment:
we can use directly dataframe.to_csv method with s3 bucket, see if we can utilize here in place of saving to disk and then uploading to s3.

Reply:
That is there for local debugging without S3, if S3_UPLOAD_CVEs is set to false. I think I'll remove the upload_s3 method and use df.to_csv with S3 if that flag is true, but keep the write to disk for debugging purposes.

Reply:
This is done.